src/ListClient.ts

```diff
@@ -100,20 +107,25 @@ export class InnerListClient<T extends any> implements ObjectClient {
     return this.clone();
   }
 
-  insertAfter(index: number, val: T): InnerListClient<T> {
-    return this.insertAt(index + 1, val);
+  insertAfter<K extends keyof T>(index: K, val: T[K]): InnerListClient<T> {
```

Comment:
similar question here, when would an index not be a number?

Reply:
^ Same as above, though this time it's more to ensure that the values are correctly interpreted.

For example:
```ts
type MyArray = [string, number];

const list = room.list<MyArray>("list")

list.get(0) // type is string 
list.get(1) // type is number
list.get(2) // type error
```

Reply:
hmm how would this array example work if I insert a new item at the beginning of the list?

Reply:
OOOOF. Yup, good point.

Reply:
Kk, fixed this, also bumped our version of typescript so we're able to do Variadic Tuple Types (https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#variadic-tuple-types)